### ISSUE TITLE
An unrecoverable has occurred in unciv while playing

### ISSUE THREAD

**Issue by Luke-Kitti** [yairm210/Unciv#7933]
**Platform**
Windows 10 (64 bit).

**Version**
I got the game on itch platform. The version of the game is 4.2.13 (build 757) 

**Describe the bug**
I was playing a vanilla ruleset with 5Hex tileset on. "An unrecoverable has occurred in unciv" while I was attacking a barbarian unit on a water tile. 

**Screenshots**
![image](https://user-images.githubusercontent.com/100391385/196887068-ccb9f8cb-298b-4f7d-8366-c2a734c1e7db.png)

**Comment by yairm210** [comment]
Please post the game save here

**Comment by Luke-Kitti** [comment]
These are the autosaves of the game I played. However, I am not sure which
one is the one that caused the bug. After the force-close window, I opened
it again and managed to finish the game without any bugs. I experienced the
same thing with a game before. However, I deleted all the saves from that
game because I reinstalled the game from the Itch platform. What I noticed
is that this never happened until I played the vanilla ruleset (I always
played the game with Rekmod).
